plugins/module_utils/podman/podman_container_lib.py

```diff
@@ -1055,8 +1051,8 @@ def compose(p, h):
         for port, hosts in ports.items():
             for h in hosts:
                 before.append(compose(port, h))
-        after = self.params['publish'] or []
-        if self.params['publish_all']:
+        after = self.params.get('publish', [])
```

Comment:
I don't think it will work so because:
```python
>>> print({'z': None}.get('z',[]))
None
```

Reply:
yea i realized this afte ri did it, i'll fix it to go back to the previous or

Reply:
But I'm not sure I understand the usecase.

Reply:
I'm directly calling PodmanManager. If you don't use the ansible module to parse all the vars, you end up with unset vars and this falls apart.  Context is https://review.opendev.org/c/openstack/tripleo-ansible/+/789956

Reply:
I'd rather extract the `update_options` and `init_options` functions to common, when you can use them probably:
```python
from ... common import init_options, update_options

options_dict = deepcopy(init_options())
options_dict = update_options(options_dict, opts)
try:
  PodmanManager(self.module, options_dict).execute()
```
and it won't require any change in the module lib. Like we do in https://github.com/containers/ansible-podman-collections/blob/95a402b8694f58b49a9194752ca47013b79effad/plugins/modules/podman_containers.py#L146-L149